components/TextField/index.jsx

```diff
@@ -155,41 +192,43 @@ export default class TextField extends Component {
       error,
       errorSubInfo,
       id,
-      inputStyle,
+      isRequired,
       label,
       labelHidden,
       maxLength,
       name,
       multiLine,
       pattern,
       placeholder,
-      isRequired,
-      rows,
       requiredLabel,
+      rows,
       title,
       type
     } = this.props;
 
     return (
-      <div className={`${formGroup} ${styles[inputStyle]} ${requiredLabel || displayLength ? styles.paddingTop : ''}`}>
-        {this.renderInputLabel()}
+      <div className={this.inputStyleClassNames}>
+        <InfoLabel
+          requiredLabel={requiredLabel}
+          inputValue={this.state.value}
+          displayLength={displayLength}
+          maxLength={maxLength} />
         {labelHidden && <span className={srOnly}>{label}</span>}
 
-        <label
-          className={`${controlLabel} ${labelHidden && hidden} ${this.hasError() ? controlLabelError : ''}`}
-          htmlFor={id}>{label}</label>
+        <label className={this.labelClassNames} htmlFor={id}>{label}</label>
 
         <div className={styles.inputContainer}>
           {
             multiLine ?
               <textarea
                 autoFocus={autoFocus}
-                className={`${formControl} ${styles.textarea} ${this.hasError() ? formControlError : ''}`}
+                className={this.textFieldClassNames}
                 disabled={disable}
                 id={id}
                 name={name}
                 maxLength={maxLength}
                 onChange={this.onChange}
+                onBlur={this.onBlur}
```

Comment:
extract the onBlur from the pros directly so you can
`onBlur={onBlur}` there is no need this have this.onBlur at all